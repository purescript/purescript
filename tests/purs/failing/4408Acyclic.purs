-- @shouldFailWith HoleInferredType
module Main where

-- Expected:
--
-- aRinku+cMuni -> bMaho -> dRei
--
-- Both aRinku and cMuni is suggested

newtype K = K Int

aRinku :: Int -> K
aRinku = K

bMaho :: K
bMaho = ?help 0

cMuni :: Int -> K
cMuni = K

dRei :: Int -> K
dRei _ = bMaho
