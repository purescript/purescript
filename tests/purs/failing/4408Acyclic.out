Error found:
in module [33mMain[0m
at tests/purs/failing/4408Acyclic.purs:16:9 - 16:14 (line 16, column 9 - line 16, column 14)

  Hole '[33mhelp[0m' has the inferred type
  [33m          [0m
  [33m  Int -> K[0m
  [33m          [0m
  You could substitute the hole with one of these values:
  [33m                                                                    [0m
  [33m  Main.aRinku                 :: Int -> K                           [0m
  [33m  Main.cMuni                  :: Int -> K                           [0m
  [33m  Safe.Coerce.coerce          :: forall a b. Coercible a b => a -> b[0m
  [33m  Unsafe.Coerce.unsafeCoerce  :: forall a b. a -> b                 [0m
  [33m  Main.K                      :: Int -> K                           [0m
  [33m                                                                    [0m

in value declaration [33mbMaho[0m

See https://github.com/purescript/documentation/blob/master/errors/HoleInferredType.md for more information,
or to contribute content related to this error.

